src/test/kotlin/org/jetbrains/research/transformations/util/ParserSetup.kt

```diff
@@ -33,35 +61,37 @@ object ParserSetup {
      * changed to "executable".
      */
     private fun makeFileExecutable(targetFile: File) {
-        LOG.info("Making parser file executable")
+        logger.info("Making parser file executable")
         val command = arrayOf("chmod", "+x", targetFile.absolutePath)
         val builder = ProcessBuilder(*command)
         builder.directory(targetFile.parentFile)
         builder.start()
     }
 
+    private fun getTmpPath(): String {
+        val tmpPath = System.getProperty("java.io.tmpdir")
+        return tmpPath.removeSuffix("/")
+    }
+
     /**
      * Checks if parser file is in the target place and it is executable.
      * if not - makes it so.
      */
     fun checkSetup() {
-        LOG.info("Checking correctness of a parser setup")
-        val pythonparserFile = File(javaClass.getResource("$PARSER_NAME.py").path)
-        val targetFile = File(TARGET_PATH)
+        logger.info("Checking correctness of a parser setup")
+        unzipParserRepo()
+        val repositoryPath = getParserRepositoryPath()
+        val pythonparserFile = File("$repositoryPath/$PARSER_RELATIVE_PATH")
+        val targetFile = File(TARGET_PARSER_PATH)
         if (!targetFile.exists() or !FileUtils.contentEquals(pythonparserFile, targetFile)) {
```

Comment:
checking their content every time is quite time-consuming, maybe we should pass the argument `toUpdate` and if so, replace the existing file with the new one? And also use it when calling unzipParserRepo()/ What do you think?

Reply:
I am not sure, because we should be sure the pythonparser file does not have an extension, has the name pythonparser and is in the tmp folder.

Reply:
Yeah, we check it when we call `targetFile.exists()`, we should do it.
I'm talking about `FileUtils.contentEquals` since I'm not sure we should check its content every time we run tests

Reply:
Sure, you are right!